src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -77,14 +74,155 @@ public void registered(SchedulerDriver driver, FrameworkID frameworkId, MasterIn
     }
     log.info("Registered framework frameworkId=" + frameworkId.getValue());
   }
-
+  
   @Override
   public void reregistered(SchedulerDriver driver, MasterInfo masterInfo) {
     log.info("Reregistered framework.");
   }
 
+  @Override
+  public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
+    log.info(String.format(
+        "Received status update for taskId=%s state=%s message='%s' stagingTasks.size=%d",
+        status.getTaskId().getValue(),
+        status.getState().toString(),
+        status.getMessage(),
+        liveState.getStagingTasksSize()));
+
+    if (!isStagingState(status)) {
+      liveState.removeStagingTask(status.getTaskId());
+    }
+
+    if (isTerminalState(status)) {
+      liveState.removeTask(status.getTaskId());
+    } else if (isRunningState(status)) {
+      liveState.updateTaskForStatus(status); // TODO(rubbish) could terminal/running get pushed to
+                                             // this
+
+      log.info(String.format("Current Acquisition Phase: %s", liveState
+          .getCurrentAcquisitionPhase().toString()));
+
+      switch (liveState.getCurrentAcquisitionPhase()) {
+        case JOURNAL_NODES :
+          if (liveState.getJournalNodeSize() == conf.getJournalNodeCount()) {
+            liveState.transitionTo(AcquisitionPhase.NAME_NODE_1);
+          }
+          break;
+        case NAME_NODE_1 :
+          if (liveState.getNameNodeSize() == 1 && liveState.getFirstNameNodeTaskId() != null) {
+            liveState.transitionTo(AcquisitionPhase.NAME_NODE_2);
+          } else {
+            log.info("Cannot locate first namenode task id");
+          }
+          break;
+        case NAME_NODE_2 :
+          if (liveState.getNameNodeSize() == HDFSConstants.TOTAL_NAME_NODES
+              && liveState.getSecondNameNodeTaskId() != null) {
```

Comment:
Seems like this currently assumes TOTAL_NN=2, since we're only checking for "SecondNNTaskId". Is there a reason to make that "Constant" configurable if we're assuming 2? Seems like things will always break when TOTAL_NN!=2
Do we want to even try to support non-HA NNs? I tend to think not.